be/src/exec/pipeline/aggregate/spillable_aggregate_blocking_source_operator.cpp

```diff
@@ -50,9 +50,19 @@ bool SpillableAggregateBlockingSourceOperator::is_finished() const {
     if (!_aggregator->spiller()->spilled()) {
         return AggregateBlockingSourceOperator::is_finished();
     }
+    if (_aggregator->spiller()->is_cancel()) {
+        return true;
+    }
     return _aggregator->is_spilled_eos() && !_has_last_chunk;
 }
 
+Status SpillableAggregateBlockingSourceOperator::set_finishing(RuntimeState* state) {
+    if (state->is_cancelled()) {
+        _aggregator->spiller()->cancel();
```

Comment:
Is it safe to call cancel here?

Reply:
> Is it safe to call cancel here?

I think it's ok. when the cancel occurs, we don’t know which pipeline will be scheduled first, the sink or the source. The purpose of the source processing is to let the spiller perceive the cancel state as soon as possible and avoid useless work. cancel itself just sets the state and will not have other effects.